Private/Get-TokenExpiry.ps1

```diff
@@ -18,10 +18,7 @@ function Get-TokenExpiry {
         [int64]$ExpiresIn
     )
     $Now = Get-Date
-    $TimeZone = Get-TimeZone
-    $UTCTime = $Now.AddSeconds($ExpiresIn)
-    $UTCOffset = $TimeZone.GetUtcOffset($(Get-Date)).TotalMinutes
-    $ExpiryDateTime = $UTCTime.AddMinutes($UTCOffset)
-    Write-Verbose "Calcuated token expiry as $($ExpiryDateTime.ToString())"
+    $ExpiryDateTime = $Now.AddSeconds($ExpiresIn)

```

Comment:
I'm -6 hours and the original logic was producing an expiration that happened in the past, causing a stack overflow.